docs/mkdocs/docs/integration/migration_guide.md

```diff
@@ -0,0 +1,248 @@
+# Migration Guide
+
+This page collects some guidelines how to future-proof your code for future versions of this library.
+
+## Replace deprecated functions
+
+The following functions have been deprecated in earlier versions and will be removed in the next major version (i.e., 
+4.0.0). All deprecations are annotated with
+[`HEDLEY_DEPRECATED_FOR`](https://nemequ.github.io/hedley/api-reference.html#HEDLEY_DEPRECATED_FOR) to report which
+function to use instead.
```

Comment:
"in earlier versions" seems weird if the functions are being deprecated in this version and is already implied by the tense.

```suggestion
The following functions have been deprecated and will be removed in the next major version (i.e., 4.0.0). All
deprecations are annotated with
[`HEDLEY_DEPRECATED_FOR`](https://nemequ.github.io/hedley/api-reference.html#HEDLEY_DEPRECATED_FOR) to report which
function to use instead.
```